lib/srv/db/opensearch/engine.go

```diff
@@ -247,30 +254,54 @@ func (e *Engine) rewriteRequest(ctx context.Context, req *http.Request) (*http.R
 	reqCopy.URL.Host = e.sessionCtx.Database.GetURI()
 	reqCopy.Host = e.sessionCtx.Database.GetURI()
 
-	return reqCopy, nil
+	return reqCopy, payload, nil
 }
 
 // emitAuditEvent writes the request and response to audit stream.
-func (e *Engine) emitAuditEvent(req *http.Request) {
-	// Try to read the body and JSON unmarshal it.
-	// If this fails, we still want to emit the rest of the event info; the request event Body is nullable,
-	// so it's ok if body is nil here.
-	body, err := libaws.UnmarshalRequestBody(req)
-	if err != nil {
-		e.Log.WithError(err).Warn("Failed to read request body as JSON, omitting the body from the audit event.")
+func (e *Engine) emitAuditEvent(req *http.Request, body []byte, statusCode uint32, noErr bool) {
+	var eventCode string
+	if noErr && statusCode != 0 {
+		eventCode = events.OpenSearchRequestCode
+	} else {
+		eventCode = events.OpenSearchRequestFailureCode
+	}
+
+	contentType := req.Header.Get("Content-Type")
+	source := req.URL.Query().Get("source")
+	if len(source) > 0 {
+		e.Log.Infof("'source' parameter found, overriding request body.")
+		body = []byte(source)
+		contentType = req.URL.Query().Get("source_content_type")
+	}
```

Comment:
Could you add a logic expelling this logic. 

Why we cannot relay of logging the whole URL with param and we need translate the url source param to the content body ?

Reply:
Technically we don't need to do this. We are not adding new information here, just parsing existing fields to ultimately extract the query category and, if possible, query text.

I'll add a comment to explain this.